### PR TITLE
CI: Updated master

### PR DESCRIPTION
**Pull request by TomAugspurger**
Syncs with daily.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
I'd like to get rid of the daily branch. Does anyone know of a way to make a kind of "conditional environment variable" in travis. Basically, we have

```
env:
  global:
      - REPO_DIR=pandas
      # on release change this to a version
      - BUILD_COMMIT=v0.24.2
```

I'd like that to be

```
env:
  global:
      - REPO_DIR=pandas
      # on release change this to a version
      - BUILD_COMMIT="master" if $TRAVIS_EVENT_TYPE=="cron" else v0.24.2
```

**Comment by jreback** [comment]
@TomAugspurger why do you need that

we could just set (after released 0.25.0);

BUILD_COMMIT=master (or empty)
and dispatch on that no?

File: .travis.yml [hunk 1]
```diff
@@ -26,65 +27,47 @@ matrix:
...
     # https://github.com/travis-ci/travis-ci/issues/8920#issuecomment-352661024
     - python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
-    - CONTAINER="wheels";
+    - CONTAINER="wheels";  # pre-relesae
     - BUILD_COMMIT=$BUILD_COMMIT;
     - BUILD_DEPENDS="$NP_BUILD_DEP Cython==0.28.2"
       # binary-only for cryptogrpahy. See https://github.com/pandas-dev/pandas/issues/26589
       # Moto picks it up, and they don't distribute 32-bit wheels.
-    - TEST_DEPENDS="$NP_TEST_DEP pytest==3.8.2 pytest-xdist pytest-mock moto hypothesis wheel==0.31.1 cryptography --only-binary=cryptography"
+    - TEST_DEPENDS="$NP_TEST_DEP pytest>=4.0.2 pytest-xdist pytest-mock moto hypothesis>=3.58 wheel==0.31.1 cryptography --only-binary=cryptography"
```
Comment (jreback) [inline .travis.yml]: we *might* need <5.0.0 here?

Comment (jreback) [reply .travis.yml]: though we are not running the slow tests so not sure.

**Comment by TomAugspurger** [comment]
Your comment made me realize an issue with my request. We may need different matrix items, deps, etc for the maintainace branch.

The best may be to do what you say, but make a `0.25.x` branch with the `0.25.0` BUILD_COMMIT, and then bug fix releases target the 0.25.x branch. Then the mental model matches up with pandas exactly.

**Comment by jreback** [comment]
what don’t u like about the current setup?
meaning the fact that we occasionally have to sync the release branch?

**Comment by TomAugspurger** [comment]
My hope was to not have daily & master drift so far apart, but I think that's unavoidable. We'll always need to have the maintenance branch (whether that's called "master" or 0.25.x) be behind.

**Comment by TomAugspurger** [comment]
All green. Merging.

**Pull request merged**